### DISCUSSION TITLE
Feature request: include in the PR closed webhook whether the PR was merged by bypassing branch protection

### DISCUSSION THREAD

**Discussion by yacinehmito** [community/community#39632] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

**The problem:** We want developers to be able to bypass branch protection for "break-glass" types of scenarios. However, in doing so, we have no way at the moment to control for the use of this bypassing feature. We would like to be notified when a PR is merged by bypassing branch protection rules. That allows us to have a record of exceptions, and check whether the use of that exception was justified.

**Suggestion solution:** Include a boolean field in the pull request event that informs of whether the PR was merged by bypassing branch protection rules.

**Temporary workaround:** At the moment, when a pull request is merged, we can fetch the branch protection rules, the review status and the check statuses, then compute the outcome of the rule ourselves to determine whether the PR was merged by bypassing branch protection.

**Comment by ossanna16** [comment]
Hi there @yacinehmito and welcome to our community! Thank you for the feedback!
